freestyle/shared/src/main/scala/freestyle/module.scala

```diff
@@ -65,24 +65,10 @@ final class moduleImpl(val c: Context) {
       cls             <- decodeAnnotationTarget(annottees.toList)
       effectTuples     = gatherEffects(cls)
       classTree       <- makeClassTree(cls)
-      objectTree      <- makeModuleTree(cls, effectTuples)
+      coproductTree    = makeCoproductTree(effectTuples)
```

Comment:
You can move the calls to `gatherEffects` and `makeCoproductTree` inside `makeModuleTree`.